File: dermacare-service/src/main/java/com/virtual/dermacare_service/service/PharmacyService.java
package com.virtual.dermacare_service.service;

import com.virtual.dermacare_service.dto.MedicineDTO;
import com.virtual.dermacare_service.dto.PrescriptionDTO;
import com.virtual.dermacare_service.exception.ResourceNotFoundException;
import com.virtual.dermacare_service.exception.UnauthorizedAccessException;

import java.util.List;

public interface PharmacyService {
    PrescriptionDTO createPrescription(PrescriptionDTO prescriptionDTO)
            throws UnauthorizedAccessException, ResourceNotFoundException;

    PrescriptionDTO getPrescription(String id)
            throws ResourceNotFoundException, UnauthorizedAccessException;

    List<PrescriptionDTO> getPatientPrescriptions(String patientId)
            throws UnauthorizedAccessException;

    String orderMedicines(String prescriptionId)
            throws ResourceNotFoundException, UnauthorizedAccessException;

    List<MedicineDTO> getAllMedicines();

    MedicineDTO addMedicine(MedicineDTO medicineDTO);

}

File: virtual-derma/src/api/pharmacyService.ts

import apiClient from './apiClient';
import { PrescriptionDTO, MedicineDTO, PrescriptionItemDTO } from '../types/pharmacyTypes';

const API_BASE_URL = '/dermacare-service/api/pharmacy';

export const createPrescription = async (
  prescriptionData: Omit<PrescriptionDTO, 'id' | 'dateCreated'>
): Promise<PrescriptionDTO> => {
  const response = await apiClient.post(`${API_BASE_URL}/prescription`, prescriptionData);
  return response.data;
};

export const getPrescription = async (id: string): Promise<PrescriptionDTO> => {
  const response = await apiClient.get(`${API_BASE_URL}/prescription/${id}`);
  return response.data;
};

export const getPatientPrescriptions = async (email: string): Promise<PrescriptionDTO[]> => {
  const response = await apiClient.get(`${API_BASE_URL}/prescription/patient/${email}`);
  return response.data;
};

export const getAllMedicines = async (): Promise<MedicineDTO[]> => {
  const response = await apiClient.get(`${API_BASE_URL}/medicines`);
  return response.data;
};

export const addMedicine = async (
  medicineData: Omit<MedicineDTO, 'id'>
): Promise<MedicineDTO> => {
  const response = await apiClient.post(`${API_BASE_URL}/medicines`, medicineData);
  return response.data;
};
